main.go

```diff
@@ -36,6 +39,44 @@ func ping(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+/* Handles a call to /books/(exected id here) and redirects depending on the requested action.  */
+func bookById(w http.ResponseWriter, r *http.Request) {
+	method := r.Method
+	if method == http.MethodGet {
+		getBookById(w, r)
+		return
+	} else {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
```

Comment:
its OK to use the `else` here, but there is no need. I would argue is better not to have it, will make the code simpler and more readable.
```suggestion

	w.WriteHeader(http.StatusMethodNotAllowed)